src/offset_of.rs

```diff
@@ -95,6 +100,32 @@ macro_rules! offset_of {
         // Compute offset.
         _memoffset_offset_from!(field_ptr, base_ptr)
     }};
+    ($parent:ty, $field:tt) => {{
+        offset_of_tuple!($parent, $field)
+    }};
+}
+
+/// Calculates the offset of the specified field from the start of the tuple.
+///
+/// ## Examples
+/// ```
+/// #[macro_use]
+/// extern crate memoffset;
+///
+/// fn main() {
+///     assert!(offset_of_tuple!((u8, u32), 1) >= 0, "Tuples do not have a defined layout");
+/// }
+/// ```
+#[macro_export(local_inner_macros)]
+macro_rules! offset_of_tuple {
+    ($parent:ty, $field:tt) => {{
+        // Get a base pointer (non-dangling if rustc supports `MaybeUninit`).
+        _memoffset__let_base_ptr!(base_ptr, $parent);
+        // Get field pointer.
+        let field_ptr = raw_field_tuple!(base_ptr, $parent, $field);
+        // Compute offset.
+        _memoffset_offset_from!(field_ptr, base_ptr)
```

Comment:
Now you are still duplicating this code. So this is no better than the previous approach, and certainly not an argument for bumping the MSRV.

We could consider bumping the MSRV if you can leave `offset_of` unchanged (besides changing `$field:tt` to something else), and only change `raw_field!` to also support tuples with the same API.

If that is not possible, please leave `offset_of!` unchanged and just add a new `offset_of_tuple!` that is used only when Rust is new enough.